spec/rubocop/cop/style/static_class_spec.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::Style::StaticClass do
+  subject(:cop) { described_class.new }
+
+  it 'registers an offense when class has only class method' do
+    expect_offense(<<~RUBY)
+      class C
+      ^^^^^^^ Prefer modules to classes with only class methods.
+        def self.class_method; end
+      end
+    RUBY
+  end
+
+  it 'registers an offense when class has `class << self` with class methods' do
+    expect_offense(<<~RUBY)
+      class C
+      ^^^^^^^ Prefer modules to classes with only class methods.
+        def self.class_method; end
+
+        class << self
+          def other_class_method; end
+        end
+      end
+    RUBY
+  end
+
+  it 'does not register an offense when class has `class << self` with macro calls' do
+    expect_no_offenses(<<~RUBY)
+      class C
+        def self.class_method; end
+
+        class << self
+          macro_method
+        end
+      end
+    RUBY
+  end
+
+  it 'registers an offense when class has assignments along with class methods' do
+    expect_offense(<<~RUBY)
+      class C
+      ^^^^^^^ Prefer modules to classes with only class methods.
+        CONST = 1
+
+        def self.class_method; end
+      end
+    RUBY
+  end
+
+  it 'does not register an offense when class has instance method' do
+    expect_no_offenses(<<~RUBY)
+      class C
+        def self.class_method; end
+
+        def instance_method; end
+      end
+    RUBY
+  end
+
+  it 'does not register an offense when class has macro-like method' do
+    expect_no_offenses(<<~RUBY)
+      class C
+        def self.class_method; end
+
+        macro_method
+      end
+    RUBY
+  end
+
+  it 'does not register an offense when class is empty' do
+    expect_no_offenses(<<~RUBY)
+      class C
+      end
+    RUBY
+  end
+
+  it 'does not register an offense when class has a parent' do
+    expect_no_offenses(<<~RUBY)
+      class C < B
+        def self.class_method; end
+      end
+    RUBY
+  end
+
+  it 'does not register an offense when class includes/extends a module' do
```

Comment:
Is it the same with `extend`? Why no offense then?

Reply:
@marcandre 
Did you mean that in the case
```ruby
class C
  extend M
end
```
we should generate an offense?

Reply:
Yes, that is what I meant. Presumably, the `extend` only adds class methods, so `C` is solely comprised of class methods and this cop should raise an offense, no?

Reply:
Agreed. Updated with the new code and test case to reflect that.